### PR TITLE
Initial import

### PR DESCRIPTION
**Pull request by port8080**

### CODE REVIEW

**Comment by port8080** [comment]
Running into the following error still, will debug with you
```
+ docker plugin enable xethub
Error response from daemon: dial unix /run/docker/plugins/71d15d4ba727e3a1546e48f1395f549618d307f89fee9e1216781e5984795aac/xethub.sock: connect: no such file or directory
```

Edit: the error response was the docker desktop needing a restart, PR is reading for your review now

File: Dockerfile.release [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+RUN mkdir -p /run/docker/plugins /mnt/state /mnt/volumes
+
+# Install dependencies
+RUN apt update && \
+    apt install -y git nfs-common && \
+    rm -rf /var/lib/apt/lists/*
+
+# Setup a default git user/email
+RUN git config --global user.name "gatling" && \
+    git config --global user.email gatling@xethub.com
```
Comment (jgodlew) [inline Dockerfile.release:14]: TODO: we could probably use docker build args to set these or have them set via options on volume creation.

File: builder/Dockerfile.rust [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# This docker image provides a way to build the binaries needed for
+# Xet services. Instead of baking the dependencies and caches into
+# the image, we leverage the host system's build cache to store the
+# intermediate artifacts using a volume for Cargo's target directory.
+FROM rust:1.70
```
Comment (jgodlew) [inline builder/Dockerfile.rust]: nit: we should use `rust:1.72` for consistency with the rest of our stack.

File: rust/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+[workspace]
+
+members = [
+    "gitxet",
```
Comment (jgodlew) [inline rust/Cargo.toml:4]: nit: For now this is fine, but in the future, we may want to have the gitxet binary be sourced from our public releases when building the plugin.

**Review by jgodlew — APPROVED** [review]
Looks good. Added some notes around future improvements to make.

**Pull request merged**